File: backend/enum-jpa-validation-app/src/main/java/com/example/enumapp/web/dto/FileAssetResponse.java
package com.example.enumapp.web.dto;

import com.example.enumapp.domain.file.FileAsset;
import com.example.enumapp.domain.file.FileAssetStatus;

import java.time.LocalDateTime;

public class FileAssetResponse {

    private Long id;
    private String originalName;
    private String contentType;
    private long sizeBytes;
    private FileAssetStatus status;
    private String ownerRef;
    private LocalDateTime createdAt;

    public static FileAssetResponse from(FileAsset asset) {
        FileAssetResponse r = new FileAssetResponse();
        r.id = asset.getId();
        r.originalName = asset.getOriginalName();
        r.contentType = asset.getContentType();
        r.sizeBytes = asset.getSizeBytes();
        r.status = asset.getStatus();
        r.ownerRef = asset.getOwnerRef();
        r.createdAt = asset.getCreatedAt();
        return r;
    }

    public Long getId() {
        return id;
    }

    public String getOriginalName() {
        return originalName;
    }

    public String getContentType() {
        return contentType;
    }

    public long getSizeBytes() {
        return sizeBytes;
    }

    public FileAssetStatus getStatus() {
        return status;
    }

    public String getOwnerRef() {
        return ownerRef;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}

File: backend/enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/web/dto/FileAssetResponse.kt
package com.example.enumapp.web.dto

import com.example.enumapp.domain.file.FileAsset
import com.example.enumapp.domain.file.FileAssetStatus
import java.time.LocalDateTime

class FileAssetResponse {
    var id: Long? = null
    var originalName: String? = null
    var contentType: String? = null
    var sizeBytes: Long = 0
    var status: FileAssetStatus? = null
    var ownerRef: String? = null
    var createdAt: LocalDateTime? = null

    companion object {
        fun from(asset: FileAsset): FileAssetResponse =
            FileAssetResponse().apply {
                id = asset.id
                originalName = asset.originalName
                contentType = asset.contentType
                sizeBytes = asset.sizeBytes
                status = asset.status
                ownerRef = asset.ownerRef
                createdAt = asset.createdAt
            }
    }
}
